src/modules/account/RadixAccount.ts

```diff
@@ -183,15 +220,97 @@ export default class RadixAccount {
     }
 
     private _onAtomReceived = async (atomObservation: RadixAtomObservation) => {
-        this.processingAtomCounter.next(this.processingAtomCounter.getValue() + 1)
+
+
+        if (!this.processingAtomCounter.closed && !this.processingAtomCounter.isStopped) {
+            this.processingAtomCounter.next(this.processingAtomCounter.getValue() + 1)
+        } else {
+            logger.error(`☢️ processingAtomCounter closed or stopped`)
+        }
 
         atomObservation.processedData = {}
         for (const system of this.accountSystems.values()) {
             await system.processAtomUpdate(atomObservation)
         }
 
-        this.processingAtomCounter.next(this.processingAtomCounter.getValue() - 1)
+        if (!this.processingAtomCounter.closed && !this.processingAtomCounter.isStopped) {
+            this.processingAtomCounter.next(this.processingAtomCounter.getValue() - 1)
+        } else {
+            logger.error(`☢️ processingAtomCounter closed or stopped`)
+        }
+
     }
 
-    
+    public async requestRadsForDevelopmentFromFaucetService(): Promise<RadixTransaction> {
+        return this.requestRadsForDevelopmentFromFaucetServiceWithoutBalanceUpdate()
+            .then((txUnique) => {
+                return this.transferSystem.getAllTransactions()
+                    .delay(2_000)
+                    .take(1)
+                    .timeout(60_000)
+                    .pipe(catchError((error, obs) => {
+                        throw new Error(`🚰 Failed to update balance after having received test XRD tokens from
+                              faucet in tx with unique string
+                               ${txUnique}, error: ${error}
+                               `)
+                    }))
+                    .pipe(map((tu: RadixTransactionUpdate) => tu.transaction))
+                    .toPromise()
+            },
+            ).catch(errorFromFaucet => {
+                const errorMsg = `Faucet server is down? ${errorFromFaucet}`
+                logger.error(errorMsg)
+                throw new Error(errorMsg)
+            })
+    }
+
+    public async requestRadsForDevelopmentFromFaucetServiceWithoutBalanceUpdate(
+        maxNumberOfRetries: number = 10,
+        killNodeServerOfFailure: boolean = true,
```

Comment:
I think this should be "killNodeServerOnFailure"

Reply:
also, why do we need that option?

Reply:
I want a `fatalError` because I dont want to wait for 150 tests to all fail very slowly, because of tokens... Just throwing an error results in slow failure of tests, but we know that almost ALL tests WILL fail, so better to kill the process and inform the user.

Makes sense?

Reply:
If this is only supposed to be used by some tests, can we put this in the test suite instead?

Reply:
It might be used by wallet for testing purposes as well, so seems like it is a good idea to not have it declared in a spec file?

Reply:
You also wrote

> If not, at least the kill process instruction can be put in the tests on error, if I understand the context correctly?

What do you mean? Code duplication..? Nah we don't want that do we? :D

Reply:
I don't see how that would be code duplication necessarily...

Reply:
sorry not convinced about this, if you still think that it's the right way, we can have a quick call maybe